src/components/sidebar/sidebar.component.js

```diff
@@ -17,33 +17,22 @@ class Sidebar extends Modal {
     );
   }
 
-  onButtonKeyDown = (ev) => {
-    if (Events.isEnterKey(ev) || Events.isSpaceKey(ev)) {
-      ev.preventDefault();
-      this.props.onCancel();
-    }
-
-    return null;
-  }
-
-  /** Returns the markup for the close icon. */
-  get closeButton() {
-    if (this.props.onCancel) {
-      return (
-        <SidebarCloseStyle>
-          <Icon
-            className='carbon-sidebar__close-icon'
-            data-element='close'
-            onClick={ this.props.onCancel }
-            type='close'
-            tabIndex='0'
-            role='button'
-            onKeyDown={ this.onButtonKeyDown }
-          />
-        </SidebarCloseStyle>
-      );
-    }
-    return null;
+  closeIcon = () => {
+    const { showCloseIcon, onCancel } = this.props;
+    if (!showCloseIcon || !onCancel) return null;
+    return (
+      <IconButton
+        onAction={ onCancel }
+      >
+        <Icon
+          type='close'
+          className='carbon-sidebar__close-icon'
```

Comment:
This PR removes some class names but has left this one in. We should be consistent with our approach.